kandji2snipe

```diff
@@ -469,31 +469,41 @@ def search_snipe_asset(serial):
 
 # Function to get all the asset models from Snipe-IT
 def get_snipe_models():
-    api_url = '{}/api/v1/models'.format(snipe_base)
-    logging.debug('Calling against: {}'.format(api_url))
-    response = requests.get(api_url, headers=snipeheaders, verify=user_args.do_not_verify_ssl, hooks={'response': request_handler})
-    if response.status_code == 200:
-        jsonresponse = response.json()
-        logging.info("Got a valid response that should have {} models.".format(jsonresponse['total']))
-        if jsonresponse['total'] <= len(jsonresponse['rows']) :
-            return jsonresponse
+    models = []
+    limit = 100
+    offset = 0
+    while True:
+        api_url = "{}/api/v1/models?limit={}&offset={}".format(
+            snipe_base, limit, offset
+        )
+
+        logging.debug(f"Calling against: {api_url}")
+
+        response = requests.get(
+            api_url,
+            headers=snipeheaders,
+            verify=user_args.do_not_verify_ssl,
+            hooks={"response": request_handler},
+        )
+        if response.status_code == 200:
+            jsonresponse = response.json()
+            logging.info(
+                "Got a valid response that should have {}/{} models.".format(
+                    len(models), jsonresponse["total"]
+                )
+            )
+            models.extend(jsonresponse["rows"])
+            offset += limit
+            if 0 == len(jsonresponse["rows"]):
```

Comment:
What would you think about making this comparison instead? `if len(jsonresponse["rows"]) < limit:`

If we're asking for 100 rows, and we only get 80, then presumably that's because there were only 80 rows left. Feels messy to make an extra API call to return 0 rows if we already know we have the lass batch, yeah?

Reply:
would be totally fine doing that comparison instead, i mostly hacked this whole thing together in a few minutes and wanted to contribute it back to other people.

Reply:
totally appreciate you sharing back! We figure anyone needing to paginate through models is using our software pretty heavily, which makes us happy! 😄